.github/workflows/CI.yml

```diff
@@ -32,6 +31,10 @@ jobs:
           # The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot
           apt install --yes --no-install-recommends ca-certificates
 
+          # register backports repository (needed for dotnet6)
+          sudo apt install  --yes --no-install-recommends software-properties-common
+          sudo add-apt-repository ppa:dotnet/backports
```

Comment:
@webwarrior-ws this is unacceptable, we don't want to use PPAs

Reply:
How are we going to install .NET 6 then?

Reply:
well, if dotnet6 is not available in 24.04 then we upgrade

Reply:
Upgraded to .NET 8